internal/component/prometheus/remote/queue/filequeue/filequeue.go

```diff
@@ -0,0 +1,182 @@
+package filequeue
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"sort"
+	"strconv"
+	"strings"
+
+	"github.com/go-kit/log"
+	"github.com/grafana/alloy/internal/component/prometheus/remote/queue/types"
+	"github.com/grafana/alloy/internal/runtime/logging/level"
+	"github.com/vladopajic/go-actor/actor"
+)
+
+var _ actor.Worker = (*queue)(nil)
+var _ types.FileStorage = (*queue)(nil)
+
+// queue represents the on disk queue. This is an ordered list by id in the format: <id>.committed
+// Each file contains a byte buffer and an optional metatdata map.
+type queue struct {
+	self      actor.Actor
+	directory string
+	maxIndex  int
+	logger    log.Logger
+	files     actor.Mailbox[types.Data]
+	// Out is where to send data when pulled from queue, it is assumed that it will
+	// block until ready for another file.
+	out func(ctx context.Context, dh types.DataHandle)
+	// existingFiles is the list of files found initially.
+	existingsFiles []string
+}
+
+// NewQueue returns a implementation of FileStorage.
+func NewQueue(directory string, out func(ctx context.Context, dh types.DataHandle), logger log.Logger) (types.FileStorage, error) {
+	err := os.MkdirAll(directory, 0777)
+	if err != nil {
+		return nil, err
+	}
+
+	// We dont actually support uncommitted but I think its good to at least have some naming to avoid parsing random files
+	// that get installed into the system.
+	matches, _ := filepath.Glob(filepath.Join(directory, "*.committed"))
+	ids := make([]int, len(matches))
+
+	// Try and grab the id from each file.
+	// 1.committed
+	for i, x := range matches {
+		id, err := strconv.Atoi(strings.ReplaceAll(filepath.Base(x), ".committed", ""))
+		if err != nil {
+			continue
+		}
+		ids[i] = id
+	}
+	sort.Ints(ids)
+	var currentIndex int
```

Comment:
```suggestion
	var currentMaxId int
```
It's not for holding an index, but for holding an ID.